src/zarr/core/array.py

```diff
@@ -977,9 +977,10 @@ def _iter_chunk_regions(
     @property
     def nbytes(self) -> int:
         """
-        The number of bytes that can be stored in this array.
+        The number of bytes that can be stored in the chunks of this array.
```

Comment:
Worth adding a note or warning that this only gives the correct answer for fixed-length types? (if I'm interpreting your TODO correctly?)

Reply:
I could also just fix the implementation ... what kind of behavior would people expect when calling `nbytes` on an array with variable length types? maybe raise an exception?

Reply:
second question, is there any reliable way to check if a dtype is variable length, other than a big lookup table?

cc @rabernat

Reply:
i'm feeling like this out of scope for this PR; numpy itself reports a (misleading) integer itemsize for variable length dtypes already, so we can lean on that bad precedent until we come up with a proper fix

Reply:
👍 - can you add a quick note to the docstring here though noting that it's incorrect for variable length dtypes?

Reply:
have a look at the docstrings I added in https://github.com/zarr-developers/zarr-python/pull/2576/commits/254e2764c1b9d64cc906e1cd9cb1dd62762a4a33